src/domains/interval.jl

```diff
@@ -244,8 +248,8 @@ function similar_interval(d::HalfLine{T,C}, a::S, b::S) where {T,S,C}
     HalfLine{promote_type(float(T),S),C}()
 end
 
-point_in_domain(d::NonnegativeRealLine) = zero(eltype(d))
-point_in_domain(d::PositiveRealLine) = one(eltype(d))
+point_in_domain(d::NonnegativeRealLine) = zero(deltype(d))
```

Comment:
Probably this name is clearer, but this is called a "choice function" in the [Axiom of Choice](https://en.wikipedia.org/wiki/Axiom_of_choice) so we could just call this `choice(d)`

Reply:
Hmm, it would be good if `choice(s)` was defined for a Set too, for example, but it isn't. This function really is just for testing purposes, but it could have any name.

Reply:
Just add `choice(d::AbstractSet) = first(d)`?

Reply:
`iterate(d)[1]` is kind of the same...

Reply:
the `point_in_domain` function roughly speaking picks some point in the interior of a domain, if possible. It may be good to think about how to "sample" points from a domain more generally (like `rand`), but that is a separate issue.

Reply:
Okay, I'm renaming to `choice` and adding a deprecation for `point_in_domain`